### PR TITLE
STRATCONN-2828 Pinterest custom data value incorrect.

### PR DESCRIPTION
**Pull request by AnkitUnthinkable**
_A summary of your pull request, including the what change you're making and why._

Changed custom data value type from number to string as Pinterest required value to be string. 
Jira tickets: https://segment.atlassian.net/browse/STRATCONN-2828?atlOrigin=eyJpIjoiYjIyNDE4YTEwNWNkNDg0ZmFhNWI5MzVhMmU0MTE5MGIiLCJwIjoiaiJ9

## Testing

Before converting custom_data value datatype to string 

Screenshot 2023-07-18 at 4 45 32 PM

After converting custom_data value data type to string

Screenshot 2023-07-18 at 4 46 12 PM

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [X] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [X] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const custom_data_field: InputField = {
       label: 'Value',
       description:
         'Total value of the event. E.g. if there are multiple items in a checkout event, value should be the total price of all items',
-      type: 'number'
+      type: 'string'
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts:17]: Will this break existing mappings?

Comment (varadarajan-tw) [reply packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts:17]: @AnkitUnthinkable - Could you confirm the impact of updating data types to existing mappings by some testing in staging?

Although Pinterest CAPI is in Public Beta, it is important to ensure that we don't break existing customer mappings.

Comment (AnkitUnthinkable) [reply packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts:17]: yes testing it on stage, I will share the status here today.

Comment (AnkitSegment) [reply packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts:17]: Mapping with value as Number are not created its throwing an error "value (123) is not a type of string". 
Already mention in Jira tickets too. 
Attached same screenshot. 

_name=Screenshot+2023-06-06+at+14 04 19

**Comment by varadarajan-tw** [comment]
@AnkitSegment – It seems like some tests are failing. Could you take a look at them?

**Review by varadarajan-tw — APPROVED** [review]
LGTM!

File: packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts (hunk 1 above)
Comment (varadarajan-tw) [reply packages/destination-actions/src/destinations/pinterest-conversions/pinterest-capi-custom-data.ts]: Discussed with @AnkitSegment! We decided not to change the type but instead do the conversion in perform block. Pinterest expects a string with number value. So, we decided to leave the type of value as number and do the string conversion internally.

**Pull request merged**